### PR TITLE
Add SO_INCOMING_CPU support to net.sock_option_flag_in

### PR DESCRIPTION
**Pull request by masaori335**
Address #5165.

Add a option (128) for `SO_INCOMING_CPU` to the `proxy.config.net.sock_option_flag_in`.

# Background

There was a bug with SO_REUSEPORT since v4.6. But it's fixed by v6.2 and some distros backported it.
Now we can use SO_INCOMING_CPU, SO_REUSEPORT and CPU Affinity 4 for perfect locality.

https://github.com/torvalds/linux/commit/b261eda84ec136240a9ca753389853a3a1bccca2

# Ref.

        SO_INCOMING_CPU (gettable since Linux 3.19, settable since Linux 4.4)

              Sets or gets the CPU affinity of a socket.  Expects an
              integer flag.

                  int cpu = 1;
                  setsockopt(fd, SOL_SOCKET, SO_INCOMING_CPU, &cpu,
                             sizeof(cpu));

              Because all of the packets for a single stream (i.e., all
              packets for the same 4-tuple) arrive on the single RX
              queue that is associated with a particular CPU, the
              typical use case is to employ one listening process per RX
              queue, with the incoming flow being handled by a listener
              on the same CPU that is handling the RX queue.  This
              provides optimal NUMA behavior and keeps CPU caches hot.
              
https://www.man7.org/linux/man-pages/man7/socket.7.html

### CODE REVIEW

File: src/iocore/net/Connection.cc [hunk 1]
```diff
@@ -238,6 +241,23 @@ Server::setup_fd_for_listen(bool non_blocking, const NetProcessor::AcceptOptions
 #endif
   }
 
+#ifdef SO_INCOMING_CPU
+  if (opt.sockopt_flags & NetVCOptions::SOCK_OPT_INCOMING_CPU) {
+    int      cpu     = 0;
+    EThread *ethread = this_ethread();
+
+#ifdef TS_USE_HWLOC
+    cpu = ethread->hwloc_obj->os_index;
```
Comment (masaori335) [inline src/iocore/net/Connection.cc:250]: The `hwloc_obj::os_index` looks "OS-provided physical index number", but hwloc has any better API?
https://github.com/open-mpi/hwloc/blob/63a8288d31a1baf67a909466aba9a022c78ca7b1/include/hwloc.h#L491

**Comment by masaori335** [comment]
[approve ci freebsd]
[approve ci debian]

**Comment by masaori335** [comment]
[approve ci freebsd]

**Comment by masaori335** [comment]
[approve ci debian]

**Comment by masaori335** [comment]
[approve ci autest]

**Comment by masaori335** [comment]
[approve ci debian]

**Review by bryancall — APPROVED** [review]

**Comment by masaori335** [comment]
@cmcfarlen I'd request to 10.0.x backport. But, this is a new feature, so we can wait for 10.1.x, if you want.

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**